src/lib/operators/get_table.cpp

```diff
@@ -23,5 +31,30 @@ std::shared_ptr<AbstractOperator> GetTable::recreate(const std::vector<AllParame
   return std::make_shared<GetTable>(_name);
 }
 
-std::shared_ptr<const Table> GetTable::_on_execute() { return StorageManager::get().get_table(_name); }
+std::shared_ptr<const Table> GetTable::_on_execute() {
+  auto original_table = StorageManager::get().get_table(_name);
+  if (!_excluded_chunks || _excluded_chunks->empty()) {
+    return original_table;
+  }
+
+  // we create a copy of the original table and don't include the excluded chunks
+  auto pruned_table = Table::create_with_layout_from(original_table, original_table->max_chunk_size());
+  auto excluded_chunks_it = _excluded_chunks->begin();
```

Comment:
In `table_scan.cpp` I convert the `vector` to an `unordered_set` and then call `excluded_chunks_set.count(chunk_id)`. I think this makes the code clearer and the chunk IDs don’t need to be sorted.